perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -467,16 +467,12 @@ public PerunBl getPerunBl() {
 
 	@Override
 	public List<Facility> getAllowedFacilities(PerunSession sess, User user) throws InternalErrorException {
-		List<Member> members = perunBl.getMembersManagerBl().getMembersByUser(sess, user);
-
-		Set<Facility> assignedFacilities = new HashSet<>();
-		for(Member member : members) {
-			if(!getPerunBl().getMembersManagerBl().haveStatus(sess, member, Status.INVALID)) {
-				assignedFacilities.addAll(this.getAssignedFacilities(sess, member));
-			}
-		}
+		return getFacilitiesManagerImpl().getAllowedFacilities(sess, user);
+	}
 
-		return new ArrayList<>(assignedFacilities);
+	@Override
+	public List<Facility> getAllowedFacilities(PerunSession sess, Member member) throws InternalErrorException {
```

Comment:
You do not have to throw InternalErrorException.

Reply:
You are right, fixed.